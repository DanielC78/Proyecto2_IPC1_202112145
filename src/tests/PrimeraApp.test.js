import { render } from "@testing-library/jest-dom";
import PrimeraApp from "../PrimeraApp";
import { shallow } from "enzyme";

describe("Prueba del componente PrimeraApp", () => {
  test("debe de mostra el componente correctamente", () => {
    const saludo = "Hola soy Goku";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar un subtitulo enviado por props", () => {
    const saludo = "Hola, soy Goku";
    const subtitulo = "soy un sub";
    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subtitulo} />
    );

    const textoParrafo = wrapper.find("p").text();
    console.log(textoParrafo);

    expect(textoParrafo).toBe(subtitulo);
  });
});
