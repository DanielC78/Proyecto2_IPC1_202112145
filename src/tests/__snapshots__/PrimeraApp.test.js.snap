// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente PrimeraApp debe de mostra el componente correctamente 1`] = `
<Fragment>
  <h1>
    Hola soy Goku
    !!!
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
